idaes/generic_models/unit_models/distillation/tray_column.py

```diff
@@ -0,0 +1,311 @@
+**default** - TemperatureSpec.none
+**Valid values:** {
+**TemperatureSpec.none** - No spec is selected,
+**TemperatureSpec.atBubblePoint** - Condenser temperature set at
+bubble point i.e. total condenser,
+**TemperatureSpec.customTemperature** - Condenser temperature at
+user specified temperature.}"""))
+    CONFIG.declare("has_heat_transfer", ConfigValue(
+        default=False,
+        domain=In([True, False]),
+        description="heat duty to/from tray construction flag.",
+        doc="""indicates if there is heat duty to/from the tray,
+**default** - False.
+**Valid values:** {
+**True** - include a heat duty term,
+**False** - exclude a heat duty term.}"""))
+    CONFIG.declare("has_pressure_change", ConfigValue(
+        default=False,
+        domain=In([True, False]),
+        description="pressure change term construction flag",
+        doc="""indicates whether terms for pressure change should be
+constructed,
+**default** - False.
+**Valid values:** {
+**True** - include pressure change terms,
+**False** - exclude pressure change terms.}"""))
+    CONFIG.declare("has_liquid_side_draw", ConfigValue(
+        default=False,
+        domain=In([True, False]),
+        description="liquid side draw construction flag.",
+        doc="""indicates if there is a liquid side draw from all trays,
+**default** - False.
+**Valid values:** {
+**True** - include a liquid side draw for all trays,
+**False** - exclude a liquid side draw for all trays.}"""))
+    CONFIG.declare("has_vapor_side_draw", ConfigValue(
+        default=False,
+        domain=In([True, False]),
+        description="vapor side draw construction flag.",
+        doc="""indicates if there is a vapor side draw from all trays,
+**default** - False.
+**Valid values:** {
+**True** - include a vapor side draw for all trays,
+**False** - exclude a vapor side draw for all trays.}"""))
+    CONFIG.declare("property_package", ConfigValue(
+        default=useDefault,
+        domain=is_physical_parameter_block,
+        description="property package to use for control volume",
+        doc="""property parameter object used to define property calculations,
+**default** - useDefault.
+**Valid values:** {
+**useDefault** - use default package from parent model or flowsheet,
+**PropertyParameterObject** - a PropertyParameterBlock object.}"""))
+    CONFIG.declare("property_package_args", ConfigBlock(
+        implicit=True,
+        description="arguments to use for constructing property packages",
+        doc="""a ConfigBlock with arguments to be passed to a property block(s)
+and used when constructing these,
+**default** - None.
+**Valid values:** {
+see property package for documentation.}"""))
+
+    def build(self):
+        """Build the model.
+
+        Args:
+            None
+        Returns:
+            None
+        """
+        # Call UnitModel.build to setup dynamics
+        super(TrayColumnData, self).build()
+
```

Comment:
Given that we allow for some critical config argument (e.g. number of trays) to have default values of `None`, I think we need to have some validation here. I agree that `None` is the correct default, but we should tell the user that they need to provide the value if they do not.

Reply:
Added a check.